### DISCUSSION TITLE
Next 13, Metadata, & ldjson?

### DISCUSSION THREAD

**Discussion by darrylmorley** [vercel/next.js#50016] [Help] [answered]
### Summary

Hi,

I'm just moving one of my sites over to v13 and was wondering how you guys are including your ldjson data now. Is there a provision for it in the new metadata function, or... how are you doing it?

**Comment by icyJoseph** [comment, marked as answer]
Shouldn't a script with strategy set to `beforeInteractive` do the trick? The MetaData API doesn't support scripts, so if I wanted to add this data:

```json
{
  "@context": "https://json-ld.org/contexts/person.jsonld",
  "@id": "http://dbpedia.org/resource/John_Lennon",
  "name": "John Lennon",
  "born": "1940-10-09",
  "spouse": "http://dbpedia.org/resource/Cynthia_Lennon"
}
```

I'd do this:

```jsx
import Script from "next/script";

export const metadata = {
  title: "Create Next App",
  description: "Generated by create next app",
};

export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return (
    <html lang="en">
      <head>
        <Script
          strategy="beforeInteractive"
          id="my-ldjson-data"
          type="application/ld+json"
          dangerouslySetInnerHTML={{
            __html: `
      {
        "@context": "https://json-ld.org/contexts/person.jsonld",
        "@id": "http://dbpedia.org/resource/John_Lennon",
        "name": "John Lennon",
        "born": "1940-10-09",
        "spouse": "http://dbpedia.org/resource/Cynthia_Lennon"
      }
      `,
          }}
        />
      </head>
      <body>{children}</body>
    </html>
  );
}

```

A couple things I don't like about this is that, without the `head` tag, a hydration error is triggered, also `beforeInteractive` is needed, even though using it should place the script in the head, but since we have a head tag around, you'd think it wouldn't be necessary.... confusing stuff, and last but not least, that the next lint config complains about using `beforeInteractive` outside `_document` -

**Reply by darrylmorley** [reply]
Similar to how I'm doing it in Next 12 then.  Not sure I had to use beforeInteractive previously. I'm surprised that there isn't a `Next` way of handling this, they seem to have covered everything else.

**Reply by M-Brico** [reply]
isn't strategy for placing scripts or loading? If it's in the head already then does it need one?

**Reply by darrylmorley** [reply]
I don't use strategy or next/script.
